src/pages/office/cat-a-mod2/office.cat-a-mod2.page.ts

```diff
@@ -0,0 +1,607 @@
+        map(data => this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE)),
+      ),
+      drivingFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        // TODO - PREP-AMOD2: update to cat a mod 2
+        map(data => this.faultCountProvider.getDrivingFaultSumCount(TestCategory.BE, data)),
+      ),
+      displayDrivingFaultComments$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.shouldDisplayDrivingFaultComments(data)),
+      ),
+      weatherConditions$: currentTest$.pipe(
+        select(getTestSummary),
+        select(getWeatherConditions),
+      ),
+      vehicleChecks$: currentTest$.pipe(
+        select(getTestData),
+        select(getVehicleChecks),
+        map(checks => [...checks.tellMeQuestions, ...checks.showMeQuestions]),
+      ),
+    };
+  }
+
+  popToRoot() {
+    const journalPage = this.navController.getViews().find(view => view.id === JOURNAL_PAGE);
+    this.navController.popTo(journalPage);
+  }
+
+  defer() {
+    this.popToRoot();
+    this.store$.dispatch(new SavingWriteUpForLater());
+    this.store$.dispatch(new PersistTests());
+  }
+
+  onSubmit() {
+    if (this.isFormValid()) {
+      this.showFinishTestModal();
+    }
+  }
+
+  identificationChanged(identification: Identification): void {
+    this.store$.dispatch(new IdentificationUsedChanged(identification));
+  }
+
+  independentDrivingChanged(independentDriving: IndependentDriving): void {
+    this.store$.dispatch(new IndependentDrivingTypeChanged(independentDriving));
+  }
+
+  weatherConditionsChanged(weatherConditions: WeatherConditions[]): void {
+    this.store$.dispatch(new WeatherConditionsChanged(weatherConditions));
+  }
+
+  routeNumberChanged(routeNumber: number) {
+    this.store$.dispatch(new RouteNumberChanged(routeNumber));
+  }
+
+  candidateDescriptionChanged(candidateDescription: string) {
+    this.store$.dispatch(new CandidateDescriptionChanged(candidateDescription));
+  }
+
+  additionalInformationChanged(additionalInformation: string): void {
+    this.store$.dispatch(new AdditionalInformationChanged(additionalInformation));
+  }
+
+  dangerousFaultCommentChanged(dangerousFaultComment: FaultSummary) {
+    if (dangerousFaultComment.source === CommentSource.SIMPLE) {
+      this.store$.dispatch(
+        new AddDangerousFaultComment(dangerousFaultComment.competencyIdentifier, dangerousFaultComment.comment),
+      );
+    } else if (startsWith(dangerousFaultComment.source, CommentSource.MANOEUVRES)) {
+      const segments = dangerousFaultComment.source.split('-');
+      const fieldName = segments[1];
+      const controlOrObservation = segments[2];
+      this.store$.dispatch(
+        new AddManoeuvreComment(
+          fieldName,
+          CompetencyOutcome.D,
+          controlOrObservation,
+          dangerousFaultComment.comment),
+      );
+
+    } else if (dangerousFaultComment.source === CommentSource.UNCOUPLE_RECOUPLE) {
```

Comment:
Are we able to remove anything to do with Uncouple/Recouple at this stage?

Reply:
We try to leave functional work for the later stories and just creat the pages now